x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/TestCaseSupplier.java

```diff
@@ -1045,6 +1056,19 @@ public static List<TypedDataSupplier> dateCases(long min, long max) {
         return cases;
     }
 
+    /**
+     *
+     * @return randomized valid date formats
+     */
+    public static List<TypedDataSupplier> dateFormatCases() {
+        return List.of(
+            new TypedDataSupplier("<format as KEYWORD>", () -> new BytesRef(ESTestCase.randomDateFormatterPattern()), DataType.KEYWORD),
```

Comment:
This only generates the named formats, as far as I can tell.  Would be good to write something that generates arbitrary format strings too, but I didn't see such a thing and this is still more coverage than we had.